### PR TITLE
Link to relevant proposal plugin from syntax plugin

### LINKED ISSUE #1949: On any syntax proposal page, recommend using the proposal plugin
Via: https://twitter.com/cassidoo/status/1085350422884245504

I thought we did this before but we don't (or it got removed when we moved the docs repo)? **EDIT**: yeah we did! https://old.babeljs.io/docs/plugins/syntax-object-rest-spread/

image

Under: https://github.com/babel/website/blob/master/docs
- https://github.com/babel/website/blob/master/docs/plugin-syntax-optional-chaining.md
- https://babeljs.io/docs/en/babel-plugin-syntax-optional-chaining

doesn't really say anything at all. We don't actually want most people to ever need to install/use these syntax plugins since what they want is the "proposal" plugins. So we should add a comment at the beginning of all of these to link to the proposal plugins that actually compile the code vs. just turning on the syntax (I've suggested removing the syntax plugins but that's another story).

If it's included in preset-env, we should mention you don't need it at all then

### PR DESCRIPTION
**Pull request by doms**
## Background

Closes #1949 

## Notes

Not entirely sure what the links to the proposal plugins should be, so I tried to follow the format of an old commit that had the original `Syntax only` markdown.

Also still awaiting a response to https://github.com/babel/website/issues/1949#issuecomment-454894305 since the files are not present in `version-7.1.0/`, but are present in `version-7.0.0`, and I'm not sure if I should be updating the `version-7.0.0/` files, or if I should add all the updated files to `version-7.1.0/`...

### CODE REVIEW

**Comment by existentialism** [comment]
@doms yep, you can update the 7.0.0 files

File: docs/plugin-syntax-throw-expressions.md [hunk 1]
```diff
@@ -4,6 +4,10 @@ title: @babel/plugin-syntax-throw-expressions
 sidebar_label: syntax-throw-expressions
 ---
 
+> #### Syntax only
+>
+> It's unlikely you want to directly use this plugin directly as it only enables Babel to parse this syntax. Instead, use [plugin-proposal-throw-expressions](babeljs.io/docs/en/plugin-proposal-throw-expressions.md) to both _parse_ and transform this syntax.
```
Comment (existentialism) [inline docs/plugin-syntax-throw-expressions.md]: Doh! I made a grammatical error... we should remove the first `directly` :(

Comment (existentialism) [reply docs/plugin-syntax-throw-expressions.md]: Also, IMO the emphasis should be around `both` and not `parse`.

**Comment by doms** [comment]
I just went through the deploy preview and realized that all the links to the proposal are broken...oops 😅 

I was linking to the actual `.md` file rather than the page that's available in the front-matter on each page.

> #### Syntax only
>
> It's unlikely you want to use this plugin directly as it only enables Babel to parse this syntax. Instead, use **--\> [not a real link](https://github.com/babel/website/pull/1951#issue-245294320) \<--** to _both_ parse and transform this syntax.

At first glance, it seems that I should be using the `id` field of the front-matter to generate the links (e.g. 
`id: babel-plugin-syntax-pipeline-operator`)

What's the correct way to handle this?

File: website/versioned_docs/version-7.0.0/plugin-syntax-do-expressions.md [hunk 2]
```diff
@@ -5,6 +5,10 @@ sidebar_label: syntax-do-expressions
 original_id: babel-plugin-syntax-do-expressions
 ---
 
+> #### Syntax only
+>
+> It's unlikely you want to use this plugin directly as it only enables Babel to parse this syntax. Instead, use [plugin-proposal-do-expressions](babeljs.io/docs/en/plugin-proposal-do-expressions.md) to _both_ parse and transform this syntax.
```
Comment (existentialism) [inline website/versioned_docs/version-7.0.0/plugin-syntax-do-expressions.md]: These should be like:

```
[plugin-proposal-do-expressions](plugin-proposal-do-expressions.md)
```

You can checkout https://raw.githubusercontent.com/babel/website/master/website/versioned_docs/version-7.0.0/plugins.md for more examples.

We want these to be relative to the version you're viewing.

Comment (existentialism) [reply website/versioned_docs/version-7.0.0/plugin-syntax-do-expressions.md]: @doms no worries, no lie... it's not really that intuitive :P

**Comment by doms** [comment]
@existentialism I just went and checked all links from the deploy preview and all seem to be working! Mind reviewing again? (oh, and thanks for being patient with me as I work to get this figured out :smile: )

**Comment by existentialism** [comment]
@doms thanks! you did great!

**Pull request merged**